akka-actor/src/main/scala/akka/actor/ActorPath.scala

```diff
@@ -87,7 +87,7 @@ object ActorPath {
   final def validatePathElement(element: String, fullPath: String): Unit = {
     def fullPathMsg = if (fullPath ne null) s""" (in path [$fullPath])""" else ""
 
-    (findInvalidPathElementCharPosition(element): @switch) match {
+    (findInvalidPathElementCharPosition(element)) match {
```

Comment:
Does this make the match less efficient for Scala 2.12 and 2.13?
I'm not sure if this is used in any hot path  but it could be since it's validating actor names. We use switch  at more places so this might become a bigger problem if switch isn't supported.

Reply:
Please note that I haven't removed all the `@switch` cases in the codebase but only 2.
In those cases the `switch` doesn't have enough cases and is not going to be optimized, Scala 2 silently accept the code while Scala 3 emits an error.

I added a comment with a note for the future, hope is enough.